### PR TITLE
2_grammar_grammar

### PR DESCRIPTION
**Pull request by psulliv**
```
>./2_grammar_grammar 
./2_grammar_grammar 
"testing terminals"
["T","not","(",")","skip","while","do","{","}",";"]
"testing nonterminals"
["cond","stmt"]
```

### CODE REVIEW

File: hw2/2_grammar_grammar.hs [hunk 1]
```diff
@@ -1 +1,84 @@
...
+-- nonterminal. A right-hand side of a production is given by a sequence of
+-- nonterminal (nt) and terminal (term) symbols.
+
+type NT = String
+
+type Term = String
+
+data Symbol = NTSymbol NT | TermSymbol Term deriving (Show)
+
+newtype Grammar = Grammar [Prod] deriving (Show)
```
Comment (caitlynplewis) [inline hw2/2_grammar_grammar.hs:22]: Any particular reason you chose newtype over just data here?

**Review by caitlynplewis — APPROVED** [review]
Not much to add here, it looks good to me

File: hw2/2_grammar_grammar.hs (hunk 1 above)
Comment (psulliv) [reply hw2/2_grammar_grammar.hs:22]: I had read about it as very similar to data and was trying it out. It only works for the case of one arg to a ctor so I wanted to see if it worked for lists.

Comment (psulliv) [reply hw2/2_grammar_grammar.hs:22]: Supposedly a little more efficient because it doesn't require a v table

**Pull request merged**